lib.sh.in

```diff
@@ -165,6 +163,12 @@ register_binfmt() {
         set_target_arch_from_platform
     fi
 
+    # In the special case where the build is native we can return
+    # without doing anything else
+    if [ "$(uname -m)" = "${XBPS_TARGET_ARCH%-musl}" ] ; then
```

Comment:
I think you can use `XBPS_ARCH` here.

Reply:
You sure? The case statement below specifically uses XBPS_TARGET_ARCH.

Reply:
I mean instead of `uname`

Reply:
oh I see. I could not tell if XBPS_ARCH is being set reliably, so I opted to use HOSTARCH instead (which is based on xbps-uhelper arch output).